### ISSUE TITLE
JPA Layer Cleanup

### ISSUE THREAD

**Issue by andrewyatz** [chanakaDe/ensembl-elastic-rest#2]
On reviewing your code I've been able to highlight a number of issues with your JPA layer and not using JPA as it is intended to be. See https://en.wikibooks.org/wiki/Java_Persistence/ManyToMany and https://en.wikibooks.org/wiki/Java_Persistence/OneToMany for more information. 

I believe these should be resolved as soon as possible.

- [ ] Excessive use of `tablename_attribute` bleeding into the application code. Rather than `GenomeJPA` having an attribute called `genome_id` it should be called `id`. JPA can handle the conversion to the schema version
- [ ] no use of `@ManyToOne` and `@OneToMany` relationships. Your work on [`Application.java`](https://github.com/chanakaDe/ensembl-elastic-rest/blob/master/src/main/java/com/chanaka/Application.java#L52), from line 52 does sub-queries to JPA for each object). If you use these attributes correctly JPA will handle the mapping. See [TrackJPA](https://github.com/andrewyatz/ensembl-elastic-rest/blob/master/src/main/java/com/chanaka/track/model/TrackJPA.java) for a full example
- [ ] use the fully resolved model now available to [convert into the Elastic model](https://github.com/andrewyatz/ensembl-elastic-rest/blob/master/src/main/java/com/chanaka/Application.java#L44). You can see here because we can use the object model as intended the conversion process is less complicated. We also remove a lot of code from [TrackJPAService}(https://github.com/andrewyatz/ensembl-elastic-rest/blob/master/src/main/java/com/chanaka/track/service/TrackJPAService.java) because there's no longer a need for it

I also have cloned this repository into [my own space](https://github.com/andrewyatz/ensembl-elastic-rest) and edited a number of classes in response to these points as examples. The main change is https://github.com/andrewyatz/ensembl-elastic-rest/commit/8339c31a0a25b910a4919d22216c43e89e4c0cbf.

Please review the changes, apply those you feel you should implement. My changes should not be merged in without you reviewing them.

**Comment by chanakaDe** [comment]
Yeah of course i will check those issues as soon as possible and let you know about the progress.
Thank you.
